docs/interpreter.md

```diff
@@ -5,11 +5,11 @@
 
 [StableHLO programs](spec_draft.md#programs) are computations over tensors
 (n-dimensional arrays), which, in the current model, are implemented using class
-`Tensor`. The underlying storage class for a `Tensor` object, `detail::Buffer`,
-stores the `mlir::ShapedType` of the tensor along with a contiguous byte array
-representing its data laid out in
-[major-to-minor order](https://www.tensorflow.org/xla/shapes). `detail::Buffer`
-objects are reference-counted to simplify memory management.
+`Tensor`, which stores the shape (an `mlir::ShapedType` object) and the
+underlying data storage (an `mlir::HeapAsmResourceBlob` object) of the tensor.
+The `mlir::HeapAsmResourceBlob` object represents a mutable blob of data laid
```

Comment:
Consider simplifying since the name of the class was just called out on the previous line. E.g. "The data storage contains a mutable blob of data...".

Reply:
done